microprofile/config/src/main/java/io/helidon/microprofile/config/ConfigCdiExtension.java

```diff
@@ -50,41 +53,65 @@
 import javax.enterprise.inject.spi.AnnotatedParameter;
 import javax.enterprise.inject.spi.Bean;
 import javax.enterprise.inject.spi.BeanManager;
-import javax.enterprise.inject.spi.CDI;
 import javax.enterprise.inject.spi.DeploymentException;
 import javax.enterprise.inject.spi.Extension;
 import javax.enterprise.inject.spi.InjectionPoint;
 import javax.enterprise.inject.spi.ProcessBean;
-import javax.enterprise.inject.spi.ProcessInjectionPoint;
 import javax.enterprise.inject.spi.ProcessObserverMethod;
-import javax.enterprise.util.AnnotationLiteral;
-import javax.enterprise.util.Nonbinding;
 import javax.inject.Provider;
-import javax.inject.Qualifier;
 
+import io.helidon.common.HelidonFeatures;
+import io.helidon.common.HelidonFlavor;
+import io.helidon.common.NativeImageHelper;
 import io.helidon.config.Config;
-import io.helidon.config.ConfigException;
 import io.helidon.config.MissingValueException;
 
 import org.eclipse.microprofile.config.ConfigProvider;
 import org.eclipse.microprofile.config.inject.ConfigProperty;
 import org.eclipse.microprofile.config.spi.ConfigSource;
 
-import static java.lang.annotation.ElementType.FIELD;
-import static java.lang.annotation.ElementType.METHOD;
-import static java.lang.annotation.ElementType.PARAMETER;
-import static java.lang.annotation.ElementType.TYPE;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
 /**
  * Extension to enable config injection in CDI container (all of {@link Config}, {@link org.eclipse.microprofile.config.Config}
  * and {@link ConfigProperty}).
  */
 public class ConfigCdiExtension implements Extension {
     private static final Logger LOGGER = Logger.getLogger(ConfigCdiExtension.class.getName());
     private static final Logger VALUE_LOGGER = Logger.getLogger(ConfigCdiExtension.class.getName() + ".VALUES");
+    private static final Annotation CONFIG_PROPERTY_LITERAL = new ConfigProperty() {
+        @Override
+        public String name() {
+            return "";
+        }
+
+        @Override
+        public String defaultValue() {
+            return UNCONFIGURED_VALUE;
+        }
+
+        @Override
+        public Class<? extends Annotation> annotationType() {
+            return ConfigProperty.class;
+        }
+    };
+
+    private static final Map<Class<?>, Class<?>> REPLACED_TYPES = new HashMap<>();
```

Comment:
Maybe some documentation?  I find as I read that I don't know what this is going to be used for.

Reply:
Documented in code. We must box primitives, as otherwise the producer does not match the type.